provider/datasource_application_dashboard_definition.go

```diff
@@ -215,27 +160,107 @@ func (a *ApplicationDashboardDefinitionDataSource) Read(ctx context.Context, req
 	response.Diagnostics.Append(diags...)
 }
 
-func getEc2AndEcsData(ctx context.Context, response *tfsdk.ReadDataSourceResponse, ecsClusterName string, ecsServiceName string) ([]builder.ElbTargetGroup, string, error) {
+func (a *ApplicationDashboardDefinitionDataSource) getResourceNames(ctx context.Context, state *ApplicationDashboardDefinitionData, response *tfsdk.ReadDataSourceResponse) (builder.ResourceNames, error) {
```

Comment:
Should we use a pointer as the first return value when we have an error in one of the following return values?
```suggestion
func (a *ApplicationDashboardDefinitionDataSource) getResourceNames(ctx context.Context, state *ApplicationDashboardDefinitionData, response *tfsdk.ReadDataSourceResponse) (*builder.ResourceNames, error) {
```